pennylane/devices/device_constructor.py

```diff
@@ -272,11 +272,18 @@ def _safe_specifier_set(version_str):
                     custom_decomps, dev
                 )
                 dev.custom_expand(custom_decomp_expand_fn)
-            else:
+            elif type(dev).preprocess != qml.devices.Device.preprocess:
                 custom_decomp_preprocess = qml.transforms.tape_expand._create_decomp_preprocessing(
                     custom_decomps, dev
                 )
                 dev.preprocess = custom_decomp_preprocess
+            else:
+                custom_decomp_preprocess_transforms = (
+                    qml.transforms.tape_expand._create_decomp_preprocess_transforms(
+                        custom_decomps, dev
+                    )
+                )
+                dev.preprocess_transforms = custom_decomp_preprocess_transforms
```

Comment:
Could we generalize here? I see that there's a lot of overlap between `_create_decomp_preprocess_transforms` and `_create_decomp_preprocessing`